llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -183,29 +183,35 @@ void PostPipeliner::computeLoopCarriedParameters() {
   }
 
   // Propagate Earliest upstream, initialize Latest
+  // Unrestricted: last cycle of last stage
+  const int Latest = NCopies * II - 1;
   for (int K = 0; K < NInstr; K++) {
     const int K2 = K + NInstr;
     const int Earliest = Info[K2].Earliest - II;
     Info[K].Earliest = std::max(Info[K].Earliest, Earliest);
-    // Unrestricted: Beyond the last stage.
-    Info[K].Latest = NCopies * II;
+    Info[K].Latest = Latest;
+    Info[K2].Latest = Latest;
```

Comment:
How can these two nodes have the same `Latest`? Is it a "modulo cycle"?

Reply:
Actually, I think that the K2 value was never used. Probably overcautious in initializing it. Earliest just defaults to zero, Latest is dependent on the length.
In a later commit we switch to a standard relative value of -1.